app/src/test/java/org/oppia/app/testing/ProfileChooserSpanTest.kt

```diff
@@ -0,0 +1,147 @@
+package org.oppia.app.testing
+
+import android.content.Context
+import android.content.res.Configuration
+import android.view.View
+import androidx.recyclerview.widget.GridLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import androidx.test.core.app.ActivityScenario.launch
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.espresso.intent.Intents
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.common.truth.Truth.assertThat
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.app.R
+import org.oppia.app.profile.ProfileChooserFragment
+import org.robolectric.annotation.Config
+
+private const val TAG_PROFILE_CHOOSER_FRAGMENT_RECYCLERVIEW = "profile_recycler_view"
+
+/**
+ * Tests for ensuring [ProfileChooserFragment] uses the correct column count for profiles based on display density.
+ * document reffered :https://developer.android.com/reference/androidx/test/core/app/ActivityScenario
+ * */
+@RunWith(AndroidJUnit4::class)
+@Config(manifest = Config.NONE)
+class ProfileChooserSpanTest {
+
+  private lateinit var context: Context
+  private var activity: ProfileChooserFragmentTestActivity? = null
+  private lateinit var fragment: ProfileChooserFragment
+  private lateinit var recyclerView: RecyclerView
+
+  @Before
+  @ExperimentalCoroutinesApi
+  fun setUp() {
+    Intents.init()
+    activity = getProfileChooserTestActivity()
+    fragment = getProfileChooserFragment()
```

Comment:
This fragment isn't technically the one you're launching later since you're initializing an activity in getProfileChooserTestActivity, then shutting it down. Instead, you should avoid caching the activity, fragment, and recycler view, and instead retrieve it locally, e.g.:

```kotlin
  @Test
  @Config(qualifiers = "land-mdpi")
  fun testProfileChooserFragmentRecyclerView_landscape_mdpi_hasCorrectSpanCount() {
    launch(ProfileChooserFragmentTestActivity::class.java).use { activity ->
      assertThat(getProfileRecyclerViewGridLayoutManager(acitivty).spanCount).isEqualTo(3)
    }
  }

  private fun getProfileRecyclerViewGridLayoutManager(activity: Activity): GridLayoutManager {
    return getProfileRecyclerView(activity).layoutManager as GridLayoutManager
  }

  private fun getProfileRecyclerView(activity: Activity): RecyclerView {
    return getProfileChooserFragment(activity).view?.findViewWithTag<View>(TAG_PROFILE_CHOOSER_FRAGMENT_RECYCLERVIEW)!! as RecyclerView
  }

  private fun getProfileChooserFragment(activity: Activity): ProfileChooserFragment {
    return activity.supportFragmentManager.findFragmentByTag(ProfileChooserFragmentTestActivity.TAG_PROFILE_CHOOSER_FRAGMENT) as ProfileChooserFragment
  }
```

Reply:
@BenHenning if we are not using fragment then why we need getProfileChooserFragment?

Reply:
Aren't we using the fragment in the above suggested code?